contentcuration/contentcuration/frontend/shared/views/channel/ChannelModal.vue

```diff
@@ -269,46 +269,81 @@
     // will never be rendered.
     beforeMount() {
       const channelId = this.$route.params.channelId;
+
+      // If there's no channelId (i.e., it's a new channel), skip verifications
+      if (!channelId) {
+        //this.isNew = true;
+        this.header = 'New Channel'; // Default header for a new channel
+        return; // Skip channel verification
+      }
+
+      // Existing channel, verify it
       return this.verifyChannel(channelId)
         .then(() => {
-          this.header = this.channel.name; // Get channel name when user enters modal
+          this.header = this.channel.name;
           this.updateTitleForPage();
           if (!this.isNew) {
             this.$refs.detailsform.validate();
           }
         })
-        .catch(() => {});
+        .catch(() => {
+          // Handle errors if needed
+        });
     },
     mounted() {
-      // Set expiry to 1ms
-      this.header = this.channel.name; // Get channel name when user enters modal
+      if (this.isNew) {
+        // For a new channel, set a default header
+        this.header = 'New Channel';
+      } else {
+        // For existing channels, use the channel's name
+        this.header = this.channel.name;
+      }
+
+      // Update the page title regardless of whether it's new or existing
       this.updateTitleForPage();
     },
+
     methods: {
       ...mapActions('channel', ['updateChannel', 'loadChannel', 'commitChannel']),
       ...mapMutations('channel', ['REMOVE_CHANNEL']),
       saveChannel() {
         this.isDisable = true;
+
         if (this.$refs.detailsform.validate()) {
           this.changed = false;
-          if (this.isNew) {
-            return this.commitChannel({ id: this.channelId, ...this.diffTracker }).then(() => {
-              // TODO: Make sure channel gets created before navigating to channel
-              window.location = window.Urls.channel(this.channelId);
-              this.isDisable = false;
-            });
-          } else {
-            return this.updateChannel({ id: this.channelId, ...this.diffTracker }).then(() => {
-              this.$store.dispatch('showSnackbarSimple', this.$tr('changesSaved'));
-              this.header = this.channel.name;
+
+          const commitOrUpdateChannel = this.isNew
+            ? this.commitChannel({ ...this.diffTracker }) // For new channel creation
+            : this.updateChannel({ id: this.channelId, ...this.diffTracker }); // For existing channel update
+
+          return commitOrUpdateChannel
+            .then(channel => {
+              if (this.isNew) {
+                const newChannelId = channel.id;
+                window.location = window.Urls.channel(newChannelId);
+                this.$store.dispatch('showSnackbarSimple', this.$tr('channelCreated'));
```

Comment:
Since the page needs a reload, I don't know that this snackbar is worth it. The display of it must race the page load, which could mean it's never shown, depending on various factors related to speed.